### DISCUSSION TITLE
32BTC Puzzle ETA?

### DISCUSSION THREAD

**Discussion by LU5GOON** [vlnahp/KeyZero#32] [General]
Hello @vlnahp 

I hope all is going well! 

I just wanted to see how far along the 32BTC puzzle commit is? And when you plan to deploy it to the project? 

I am really looking forward to using it!

Have a great day!

LU5GOON

**Comment by vlnahp** [comment]
Hi!
Currently, I'm busy doing my university assignments. 
You can temporarily use this address and range: `range: 9223372036854775808-18446744073709551615`, clear the `address.txt` and put this address in it `address: 16jY7qLJnxb7CHZyqBP8qca9d51gAjyXQN`. Remember to choose sequential offline brute.
0.64BTC is expecting you! 😄

**Reply by LU5GOON** [reply]
Hello @vlnahp 

Thank you so much for getting back to me. What would I need to type as the command to run the ranges? I understand the address.txt part but what would I need to enter in your program to perform this technique?

Sorry for the Nooby question!!

**Reply by LU5GOON** [reply]
Never mind I am silly. I love your program it's easy to use.

Thank you for the feature to get started with! Can't wait to see what other updates your produce down the line!

**Reply by vlnahp** [reply]
I'm so happy that Btcbf has such curious users!
Just delete the `cache.txt` and do this 
```shell 
$ python3 Btcbf.py 

 What do you want to do? 
 
   [1]: generate random key pair 
   [2]: generate public address from private key 
   [3]: brute force bitcoin offline mode 
   [4]: brute force bitcoin online mode 
   [0]: exit 
 
 Type something>3
 
 Enter the desired number: 
 
   [1]: random attack 
   [2]: sequential attack 
   [0]: exit 
 
 Type something>2

 Enter range (example:1-100)>9223372036854775808-18446744073709551615

Number of available cores: 2
 
 How many cores to be used? (leave empty to use all available cores) 
 
 Type something>1

```

**Reply by LU5GOON** [reply]
This is fantastic, thank you!

I noticed after deleting the cache when you run it, it creates another. 

Does this mean it remembers where I last stopped on so it can resume the next time I run it from where I was previous? 

This would be great if so, and great if someone turns their PC off from time to time! 

Thank you 😁😁

**Reply by vlnahp** [reply]
Yes, it is exactly meant to do this.

**Reply by LU5GOON** [reply]
Thank you this has been great! I am 26 hours and counting with the ongoing puzzle solve. You say our chances are higher, is there a estimated time frame to crack it or estimation from other successful people you may of spoken to or witnessed?

**Comment by LU5GOON** [comment]
Thank you this has been great! I am 26 hours and counting with the ongoing puzzle solve. You say our chances are higher, is there a estimated time frame to crack it?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by vlnahp** [reply]
totalRange=18446744073709551615-9223372036854775808=9223372036854775807
speed=6000keys/second
estimatedDays=totalRange/(speed*3600*24)=9223372036854775807/(6000*3600*24)=17791998527days (Yes, this is Bitcoin!!!🙃)
estimatedYears=totalRange/(speed*3600*24*365)=9223372036854775807/(6000*3600*24*365)=48745201 (Crazy!💀)
Yes, I said we have more chance this way, but still it's too low. Millions of years against Billions of Billions of (and Billions of ...) years.

**Reply by LU5GOON** [reply]
Hey man, i just wanted to reach out regarding the BTC puzzle cache file. I have been on and off with letting it try solve but curious if i am getting anywhere as the numbers in the cache file have not changed from day 1.

I just want to make sure i am not going back to the start if the numbers in the cache file have not updated
Can you confirm it does in fact continue where i left off each time? i have never deleted the cache file

Cache still says & stays at: 9223372038612754490-0-18446744073709551615

hope to hear back from you soon :)

@vlnahp

**Reply by vlnahp** [reply]
I'll review the code and check it soon. Sorry for the delay.

**Reply by LU5GOON** [reply]
> I'll review the code and check it soon. Sorry for the delay. 

Thank you man, I have devoted to it and want to make sure it's actually progressing every time I stop and start again another day.

**Reply by vlnahp** [reply]
Hey @LU5GOON! I checked it. Everything works seamlessly. I also fixed some bugs. Backup the `cache.txt` and clone the latest version of **_Btcbf_.**

**Reply by LU5GOON** [reply]
> Hey @LU5GOON! I checked it. Everything works seamlessly. I also fixed some bugs. Backup the `cache.txt` and clone the latest version of **_Btcbf_.**

Thank you so much @vlnahp so it does continue on from where I left off? Thank you for checking and yep I will keep the current cache and Download latest release! Thanking you!
